From smurf@smurf.noris.de Wed Jul 12 23:56:33 2006
From: Matthias Urlichs <smurf@smurf.noris.de>
To: gregkh@suse.de
Cc: Matthias Urlichs <smurf@smurf.noris.de>
Subject: USB: Option driver: Short driver names were identical
Message-ID: <E1G0v6i-0001DC-Lr@smurf.noris.de>

The short driver names were not unique,
which prevented the driver from actually loading.

Also, one of the ioctl pointers was missing.

Signed-Off-By: Matthias Urlichs <smurf@smurf.noris.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/option.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- gregkh-2.6.orig/drivers/usb/serial/option.c
+++ gregkh-2.6/drivers/usb/serial/option.c
@@ -128,7 +128,7 @@ static struct usb_driver option_driver =
 static struct usb_serial_driver option_3port_device = {
 	.driver = {
 		.owner =	THIS_MODULE,
-		.name =		"option",
+		.name =		"option3",
 	},
 	.description       = "GSM modem (3-port)",
 	.id_table          = option_ids3,
@@ -143,6 +143,7 @@ static struct usb_serial_driver option_3
 	.chars_in_buffer   = option_chars_in_buffer,
 	.throttle          = option_rx_throttle,
 	.unthrottle        = option_rx_unthrottle,
+	.ioctl             = option_ioctl,
 	.set_termios       = option_set_termios,
 	.break_ctl         = option_break_ctl,
 	.tiocmget          = option_tiocmget,
@@ -155,7 +156,7 @@ static struct usb_serial_driver option_3
 static struct usb_serial_driver option_1port_device = {
 	.driver = {
 		.owner =	THIS_MODULE,
-		.name =		"option",
+		.name =		"option1",
 	},
 	.description       = "GSM modem (1-port)",
 	.id_table          = option_ids1,
@@ -630,7 +631,6 @@ static void option_setup_urbs(struct usb
 
 	dbg("%s", __FUNCTION__);
 
-
 	for (i = 0; i < serial->num_ports; i++) {
 		port = serial->port[i];
 		portdata = usb_get_serial_port_data(port);
